minari/dataset/minari_dataset.py

```diff
@@ -253,7 +201,7 @@ def filter_episodes(self, condition: Callable[[h5py.Group], bool]) -> MinariData
         ```
 
         Args:
-            condition (Callable[[h5py.Group], bool]): callable that accepts an episode group and returns True if certain condition is met.
```

Comment:
This relaxes the type expected as an input for the filter function to be applied to the underlying episode dataset to be any type, since MinariDatset is no longer supposed to assume the types used in `MinariStorage`

Reply:
How about EpisodeData?

Reply:
Right now `MinariStorage.apply()` calls the filter function with `h5py.group` representations of episodes. We'd need to make it apply pass  `EpisodeData` into the filter function instead of `h5py.group` Curious to hear your thoughts on this.

Reply:
From the user perspective, it makes much more sense to have `EpisodeData`; however this means that we should convert the `h5py.group` into `EpisodeData` inside the function. 
This means we add a dependency `MinariStorage` -> `EpisodeData`, and now it makes sense to also change the return type of `get_episodes` to be already `EpisodeData`.

The other alternative is to have `dict`; I think this is more weird for the user, because it is not immediately clear what keys you can use. On the other side it gives more flexibility for possible future `MinariStorage`.

A solution in the middle is having `EpisodeData` in `MinariDataset` and `dict` in `MinariStorage`; then you need to wrap the function in `MinariDataset` to convert the `dict` to `EpsiodeData`. It complicates a bit more the code, but it has the advantage of both. 

I am towards the third option, but I don't have a strong opinion on this; feel free to choose one of the three if you see something else.

Reply:
apply actually gives the condition function a h5py group rather than a dict. Do you think we should change apply to give dicts?

Reply:
I think I understand, I'm tentatively changing apply to give the condition function `dict`s and appropriately wrapping the condition function in `filter_episodes`